src/Nix/Delegate.hs

```diff
@@ -260,24 +260,34 @@ exchangeKeys key host = do
 
           liftIO (Control.Exception.handle handler1 download)
 
-          -- NB: path shouldn't is a FilePath and won't have any
-          -- newlines, so this should be okay
-          Turtle.err (Turtle.unsafeTextToLine $ Turtle.format ("[+] Installing: "%fp) path)
-
-          warnSudo
-
-          let install =
-                  Turtle.procs "sudo"
-                      [ "mv"
-                      , Turtle.format fp localPath
-                      , Turtle.format fp path
-                      ]
-                      empty
-          let handler2 :: SomeException -> IO ()
-              handler2 e = do
-                  let pathText      = Turtle.format fp path
-                  let exceptionText = Data.Text.pack (show e)
-                  let msg           = [NeatInterpolation.text|
+          ec <- Turtle.proc "cmp"
```

Comment:
The main thing I'd suggest is comparing the files in Haskell instead of using an external process

Reply:
OK, please see 76d5c7ad1279a6530331b1a4302a378bf9836939 .